cql3/expr/expression.hh

```diff
@@ -642,7 +642,7 @@ inline bool is_multi_column(const binary_operator& op) {
     return expr::is<tuple_constructor>(op.lhs);
 }
 
-inline bool has_token(const expression& e) {
+inline bool has_partition_token(const expression& e, const schema& table_schema) {
```

Comment:
Added a comment.

I agree that the name `has_` is a bit unfortunate, it suggests that we do a search over all the expression AST nodes.

I think it originally came from a statement_restrictions method that was named in the same way.

My hope is that it will later disappear when I get to rewriting `statement_restrictions`.
Right now it's used to see if the restrictions are `partition_token = ...`, but it's a bit fragile, I think query analysis should generally be changed to rely more on pattern matching, and less on random functions and allowing only a strict subset of restrictions to be written.